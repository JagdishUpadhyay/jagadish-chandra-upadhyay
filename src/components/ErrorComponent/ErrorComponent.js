import React from "react";
import PropTypes from "prop-types";
import "../../assets/styles/error.scss";

/**
 * ErrorComponent to display error to user.
 * @param {object} props 
 */
export const ErrorComponent = (props) => {
    return (
        <div className="error">{props.errorMsg}</div>
    );
}

ErrorComponent.propTypes = {
    /**
     * error message to display to user.
     */
    errorMsg: PropTypes.string
}

ErrorComponent.defaultProps = {
    errorMsg: "Something went wrong. Please try again later."
}
