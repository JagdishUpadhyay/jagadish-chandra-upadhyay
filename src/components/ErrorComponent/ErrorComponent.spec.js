import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import { ErrorComponent } from "../ErrorComponent/ErrorComponent";

describe("ErrorComponent", () => {
    let component;

    beforeEach(() => {
        component = shallow(<ErrorComponent />);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(component)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('error class should be there', () => {
        expect(component.find('.error')).toHaveLength(1);
    });

    it('should render default message if errorMsg is not passed', () =>{
        expect(component.find('.error').text()).toEqual("Something went wrong. Please try again later.");
    });
});
