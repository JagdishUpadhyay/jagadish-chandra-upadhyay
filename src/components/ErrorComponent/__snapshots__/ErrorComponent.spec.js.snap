// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent renders correctly 1`] = `
<div
  className="error"
>
  Something went wrong. Please try again later.
</div>
`;
